Packs/AWS-CloudTrail/Integrations/AWS-CloudTrail/AWS-CloudTrail.py

```diff
@@ -412,29 +413,113 @@ def test_function():
         demisto.results('ok')
 
 
+def calculate_fetch_start_time(last_fetch, first_fetch):
+    #if last_fetch is None:
+    if not last_fetch:
+        if not first_fetch:
+            first_fetch = '5 days'
+        first_fetch_dt = parse(first_fetch).replace(tzinfo=utc)
+        first_fetch_time = first_fetch_dt.timestamp()
+        return first_fetch_time
+    else:
+        return float(last_fetch)
+
+def fetch_incidents(args: dict, params: dict):
+    client = aws_session(
+         region=args.get('region'),
+         roleArn=args.get('roleArn'),
+         roleSessionName=args.get('roleSessionName'),
+         roleSessionDuration=args.get('roleSessionDuration'),
+    )
+
+    last_fetch = demisto.getLastRun()
+    demisto.debug(f"Last Run: {last_fetch}")
+    first_fetch = params.get('first_fetch')
+    attribute_key = params.get('AttributeKey')
+    if not attribute_key:
+        attribute_key = 'EventName'
+    attribute_value = params.get('AttributeValue')
+
+    fetch_limit = int(params.get('fetch_limit'))
+    if fetch_limit > 50 or fetch_limit <= 0:
+        fetch_limit = 50
+
+    fetch_start_time = calculate_fetch_start_time(last_fetch, first_fetch)
+    demisto.debug("Fetch start time")
+    demisto.debug(str(fetch_start_time))
+
+    incidents = []
+    incident_created_time = fetch_start_time
+    kwargs = {
+        'LookupAttributes': [{
+            'AttributeKey': attribute_key,
+            'AttributeValue': attribute_value
+        }]
+    }
+
+    kwargs.update({'StartTime': fetch_start_time})
+    #demisto.log("These are the arguments: ")
+    #demisto.log(str(kwargs))
+
```

Comment:
```suggestion
```

Reply:
Yes these were for testing only. I took them off